Files.Launcher/Program.cs

```diff
@@ -233,17 +239,32 @@ private static async Task parseArguments(AppServiceRequestReceivedEventArgs args
                     process.Start();
                     break;
 
-                case "LoadMUIVerb":
-                    var responseSet = new ValueSet();
-                    responseSet.Add("MUIVerbString", Win32API.ExtractStringFromDLL((string)args.Request.Message["MUIVerbLocation"], (int)args.Request.Message["MUIVerbLine"]));
-                    await args.Request.SendResponseAsync(responseSet);
+                case "LoadContextMenu":
+                    var contextMenuResponse = new ValueSet();
+                    var loadThreadWithMessageQueue = new Win32API.ThreadWithMessageQueue<ValueSet>(HandleMenuMessage);
+                    var cMenuLoad = await loadThreadWithMessageQueue.PostMessage<Win32API.ContextMenu>(args.Request.Message);
+                    contextMenuResponse.Add("Handle", handleTable.AddValue(loadThreadWithMessageQueue));
+                    contextMenuResponse.Add("ContextMenu", JsonConvert.SerializeObject(cMenuLoad));
+                    await args.Request.SendResponseAsync(contextMenuResponse);
                     break;
 
-                case "ParseAguments":
-                    var responseArray = new ValueSet();
-                    var resultArgument = Win32API.CommandLineToArgs((string)args.Request.Message["Command"]);
-                    responseArray.Add("ParsedArguments", JsonConvert.SerializeObject(resultArgument));
-                    await args.Request.SendResponseAsync(responseArray);
+                case "ExecAndCloseContextMenu":
+                    var menuKey = (string)args.Request.Message["Handle"];
+                    var execThreadWithMessageQueue = handleTable.GetValue<Win32API.ThreadWithMessageQueue<ValueSet>>(menuKey);
+                    if (execThreadWithMessageQueue != null)
+                    {
+                        await execThreadWithMessageQueue.PostMessage(args.Request.Message);
+                    }
+                    //handleTable.RemoveValue(menuKey);
```

Comment:
Could it be removed?

Reply:
Do you mean the commented line? I'd like to keep it: that call is needed cleanup resources when menu is closed. Unfortunately if you uncomment it some menu items will randomly stop working.
Resource cleanup is currently done on app closing, but I hope to find a solution for the issue above and cleanup as soon as a menu is closed.
I'll add a comment about this.